onnxscript/rewriter/_rewrite_rule.py

```diff
@@ -136,7 +136,71 @@ def match(
                 if var.name is not None:
                     if var.name not in match.bindings:
                         match.bind(var.name, None)
+            
+            # Perform value/node level checks before condition function
             try:
+                # Check node-level checkers
+                for pattern_node, ir_node in match.node_bindings.items():
+                    if hasattr(pattern_node, '_check') and pattern_node._check is not None:
```

Comment:
The hasattr check should be unnecessary given that the object is a pattern node.